### PR TITLE
FIX issue #4852 Dockerfile does not build with 0.12.3 

### PR DESCRIPTION
**Pull request by skalahonza**
I edited the Dockerfile so it can be acutally built. I also added a docker-compose file so you can run nightscout and mongodb locally using docker-compose up. 

This is also related to 
* Azure deploy fail #5640
* build official docker image for each new release #5503

Because it is now possible to run Docker images pushed to DockerHub in the Azure AppService.

## Building official Docker image
It is now possible to let DockerHub build image for you. All you need to do is to link it with the official repository.
https://docs.docker.com/docker-hub/builds/

You can build images with tag dev from dev branch and official latest/production images from master branch with ease.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    environment:
+      API_SECRET: "${API_SECRET:-123456789abc}"
+      MONGO_CONNECTION: 'mongodb://${MONGODB_USERNAME:-nightscout}:${MONGODB_PASSWORD:-secret}@database/nightscout?retryWrites=true&w=majority&authSource=admin'
+      MONGO_COLLECTION: 'entries'
+      INSECURE_USE_HTTP: "true"
+      SECURE_HSTS_HEADER: "false"
+      PORT: 1337
+      TIME_FORMAT: "24"
+      DISPLAY_UNITS: "mmol"
+      WAIT_HOSTS: "database:27017"
```
Comment (m-a-v) [inline docker-compose.yml:21]: This doesn't work for me. I had to use something like

`command: sh -c "sleep 5; node server.js"`

Comment (skalahonza) [reply docker-compose.yml:21]: Can you pls post the std out output in the comments? It works fine on my machine.

Comment (m-a-v) [reply docker-compose.yml:21]: Actually I am not using your branch, but my docker compose file is very similar. I've also added `WAIT_HOSTS`. I think it is a timing problem. As you can see in the startup sequence the nightscout-app is not waiting long enough. Although nightscout-app makes afterward a successful connection, Nightscout doesn't work. I didn't have this problem with the same docker-compose.yml file before upgrading to 14.0.7.

```
nightscout-db | 2020-11-27T16:01:00.060+0000 I CONTROL  [initandlisten] MongoDB starting : pid=1 port=27017 dbpath=/data/db 64-bit host=a6370c7791ae

...

nightscout-app | Error connecting to MongoDB: {"name":"MongoNetworkError"} - retrying in 3 sec
nightscout-app | ERROR CONNECTING TO MONGO Error: MongoDB connection failed! Double check the MONGODB_URI setting in Heroku.
nightscout-app |     at MongoClient.connect.then.catch.err (/opt/app/lib/storage/mongo-storage.js:75:30)
nightscout-app |     at process._tickCallback (internal/process/next_tick.js:68:7)
nightscout-app | Mongo Storage system ready

...

nightscout-db | 2020-11-27T16:01:01.177+0000 I NETWORK  [initandlisten] waiting for connections on port 27017

...

nightscout-app | Successfully established a connected to MongoDB
nightscout-app | Mongo user role seems ok: [ { role: 'root', db: 'admin' } ]
nightscout-app | Mongo Storage system ready

```

Comment (skalahonza) [reply docker-compose.yml:21]: WAIT_HOSTS is not a built in functionality, I had to add support for it in the Dockerfile
```
## Add the wait script to the image
ADD https://github.com/ufoscout/docker-compose-wait/releases/download/2.7.3/wait /wait
RUN chmod +x /wait
```

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+version: "3.4"
+
+volumes:
+  mongodb_data_container:
+
+services:
+  nightscout:
+    image: "skalahonza/nightscout"
```
Comment (m-a-v) [inline docker-compose.yml:8]: This seems to be obsolete and should be discarded.

Comment (skalahonza) [reply docker-compose.yml]: Good point. Discarded.

**Comment by sulkaharo** [comment]
Apologies for the long latency - DEV now has working Docker setup, coupled with Github Actions Docker build. More improvements to land soon. https://hub.docker.com/r/nightscout/cgm-remote-monitor/tags?page=1&ordering=last_updated

**Comment by m-a-v** [comment]
@sulkaharo No worries for the long latency. I have running a similar setup like in this PR for several months. I think it would be important also to add a docker compose solution for the people. If you want to I could review and test @skalahonza in more detail.

**Comment by sulkaharo** [comment]
@pazaan is working on the compose - we were discussing supporting both compositions that only contain the Nightscout Node component but also one that has Mongo as well, which would be a fully self contained image.

**Comment by m-a-v** [comment]
@pazaan Is there some progress for docker compose?

**Comment by pazaan** [comment]
See https://github.com/nightscout/cgm-remote-monitor/pull/6903

**Pull request closed without merging**